File: go/gin/internal/modules/admin/controller.go
package admin

import (
	"net/http"
	"strconv"
	"tcc/internal/common/errors"
	"tcc/internal/common/helper"
	"tcc/internal/common/messages"
	"tcc/internal/modules/admin/dto"

	"github.com/gin-gonic/gin"
)

type AdminController struct{
	service AdminService
}

func (c *AdminController) FindAllManagers(ctx *gin.Context){
	quantidade := ctx.Query("quantidade")

	quantidadeInt, _ := strconv.Atoi(quantidade)

	dto := dto.FindAllQueryDto{
		Quantidade: quantidadeInt,
	}

	if err:= helper.Validate(dto); err != nil {
		badRequest := errors.BadRequestError(messages.ErrorMessages.BadRequest, err...)
		ctx.JSON(badRequest.StatusCode, badRequest)
		return 
	}


	managers, err := c.service.FindAllManagers(dto.Quantidade)
	if err != nil {
		ctx.JSON(err.StatusCode, err)
		return
	}

	ctx.JSON(http.StatusOK, managers)
}

func (c *AdminController) FindAllClients(ctx *gin.Context){
	quantidade := ctx.Query("quantidade")

	quantidadeInt, _ := strconv.Atoi(quantidade)

	dto := dto.FindAllQueryDto{
		Quantidade: quantidadeInt,
	}

	if err:= helper.Validate(dto); err != nil {
		badRequest := errors.BadRequestError(messages.ErrorMessages.BadRequest, err...)
		ctx.JSON(badRequest.StatusCode, badRequest)
		return 
	}


	clients, err := c.service.FindAllClients(dto.Quantidade)
	if err != nil {
		ctx.JSON(err.StatusCode, err)
		return
	}

	ctx.JSON(http.StatusOK, clients)
}

File: rust/actix/src/modules/admin/controller.rs
use actix_web::{
    web::Query,
    HttpResponse
};
use validator::Validate;
use crate::common::errors::bad_request::bad_request_error;
use crate::common::messages::error::ErrorMessages;
use crate::common::errors::base::BaseError;

use super::dto::FindAllQueryDto;
use super::service::AdminService;

pub struct AdminController{
}

impl AdminController {
    pub async fn find_all_managers(
        dto: Query<FindAllQueryDto>
    ) -> Result<HttpResponse, BaseError> {
        if let Err(err) = dto.validate() {
            return Err(bad_request_error(ErrorMessages::BAD_REQUEST, Some(err)));
        }

        match AdminService::find_all_managers(dto.quantidade).await {
            Ok(managers) => Ok(HttpResponse::Ok().json(managers)),
            Err(err) => Err(err),
        }
    }

    pub async fn find_all_clients(
        dto: Query<FindAllQueryDto>
    ) -> Result<HttpResponse, BaseError> {
        if let Err(err) = dto.validate() {
            return Err(bad_request_error(ErrorMessages::BAD_REQUEST, Some(err)));
        }

        match AdminService::find_all_clients(dto.quantidade).await {
            Ok(clients) => Ok(HttpResponse::Ok().json(clients)),
            Err(err) => Err(err),
        }
    }
}
